### PR TITLE
[CIRCLE-23202] Add new cluster creation jobs and integration tests [semver:major]

### PR DESCRIPTION
**Pull request by lokst**
Ticket: [CIRCLE-23202]
Changes:
- Destructure orb.yml
- Add new commands, jobs and examples to support cluster and node pool creation and deletion
- Add integration tests including for Windows node pool (note: Windows node pool creation requires project to be whitelisted under early access program) Windows support tested here: https://app.circleci.com/jobs/github/CircleCI-Public/gcp-gke-orb/108/parallel-runs/0/steps/0-114
- Replace `init` command with `update-kubeconfig-with-credentials` command
- Use new orb-tools workflow (`circleci/orb-tools@dev:newworkflow`, to be replaced with release version once available)
- Fix `image` passed to `rollout-image` command by `publish-and-rollout-image` job. Resolves https://github.com/CircleCI-Public/gcp-gke-orb/issues/1
- Add `tag` parameter to `rollout-image` command. Resolves https://github.com/CircleCI-Public/gcp-gke-orb/issues/5
- Update version of `gcp-cli`, `gcp-gcr` and `kubernetes` orb dependencies

[CIRCLE-23202]: https://circleci.atlassian.net/browse/CIRCLE-23202

### CODE REVIEW

**Comment by eddiewebb** [comment]
Is the reason for `major` the change of command names?

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,43 +1,221 @@
 version: 2.1
 
+orb_promotion_filters: &orb_promotion_filters
+  branches:
+    ignore: /.*/
+  tags:
+    only: /^(major|minor|patch)-release-v\d+\.\d+\.\d+$/
```
Comment (eddiewebb) [inline .circleci/config.yml]: how do we know the version before it is promoted?

Comment (lokst) [reply .circleci/config.yml]: @eddiewebb The version numbering must be manually derived based on the last published number. The numbering used in the tag is actually irrelevant to the CLI promotion command, but my thoughts were that this tagging pattern might be useful for recordkeeping purposes (e.g. for those who use GitHub releases) and also keeps the tags unique.

**Comment by lokst** [comment]
> This is a very large collecton of changes. Should we isolate some of them?

@eddiewebb Agreed; due to the timeline and holiday season, I had made one PR, but I could rework these changes into a few PRs instead if that's better?

**Comment by lokst** [comment]
> Is the reason for `major` the change of command names?

Yes, the main breaking changes are:
- removal of the `init` command
- some renamed parameters in `rollout-image` command and `publish-and-rollout-image` job due to upgrade of `circleci/gcp-gcr` orb dependency

Changes that may have unexpected outcomes for some users
- change in implementation of `rollout-image` command (now sets the container image to <<parameters.image>>:<<parameters.tag>> instead of <<parameters.image>>)
- usage of docker executor by default in `publish-and-rollout-image` job instead of `machine`
- `install` command now does not install `kops` as it now uses the kubernetes orb's `install-kubectl` command instead of `install`

File: .circleci/config.yml (hunk 1 above)
Comment (eddiewebb) [reply .circleci/config.yml]: got it, thanks

**Pull request merged**